src/polycubed/src/rest_server.h

```diff
@@ -153,6 +164,29 @@ class RestServer {
   static std::string blacklist_cert_path;
 
   std::shared_ptr<spdlog::logger> logger;
+
+  // the struct Metric for now is composed of two vectors for each metric (Gauge
+  // and Counter), one indicates the family of the metric and one the metric
+  // itself Note: families combine values with the same name, but distinct label dimensions
+  // TODO maybe there are best ways
+  struct Metric {
```

Comment:
Hi @FedeParola , regarding the first part I think it is better to keep the two functions separate. The ```RestServer::create_metrics()``` function reads the data deriving from the parsing of the yangs and creates the data structures. ```RestServer::get_metrics()``` uses these structures only to add the values ​​I get with jsoncons. I don't delete any cube, I just do checks:
1. does the cube exist and do I have the metric? Then I update the value
2. does the cube exist and I don't have the metric? then I add a label that is cubeName = "cubeName"
3. do I have the metric of a cube that has been cleared? then I have to clear all his metrics. This is because if you create another cube of the same service with the same name on a logical level, they are two different cubes and therefore two different stories, but if you do not delete it, it would result as if it were the same cube.

Another reason why I thought about dividing the two parts is that Prometheus does the endpoint scraping which is assigned every 15 seconds so I save time by not recreating the data structures.

Tell me though if I misunderstood something from your comment :muscle:

Reply:
I understand, thanks for your clarification. What about `struct Metric`? Do you think it would be possible to simplify it to only include two maps? I mean, now you are using a vector to store the metrics and a map to perform the mapping MetricName -> PositionInTheVector, but why not storing directly the metric in a map indexed by the MetricName?

Reply:
Hi @FedeParola , as far as struct Metric is concerned. It is made up of two vectors and two maps simply because I started following the examples of Prometheus and in the meantime I was adapting the code to my needs. Yesterday I had some time to think about it and I modified the struct and also the rest server. This is the struct of now:

```
  struct Metric {
    std::map<std::string, std::reference_wrapper<prometheus::Family<prometheus::Counter>>> counters_map;
    std::map<std::string, std::reference_wrapper<prometheus::Family<prometheus::Gauge>>> gauges_map;
  };
```

In the rest server in terms of performance, not much has improved since I still have to cycle through all the metrics of each cube, but in doing so I saved something in terms of data structures, so thanks again for your observation :muscle: .

Reply:
Great! I know there isn't a performance improvement but this change should make the code easier to understand.